### PR TITLE
Issue #5: Feature/made the feature card receive the feature from features component

### LINKED ISSUE #5: User would like to view features
- [x] User should be able to view features

- [x] User should be able to paginate

- [ ] User should be able to search

### PR DESCRIPTION
**Pull request by J-DK**
# Description

The feature-card component shows the feature which is defined in itself. 
Now the feature is being sent from the features component (not from the service as it gives 404 error).
The actual feature of _adding a feature_ is not done yet. (in progress)

Fixes #5 

## Type of change

Please check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by paul58914080** [comment]
Hello

You can start your server with

https://github.com/paul58914080/ff4j-web-console/blob/master/package.json#L13

`npm run start:proxy:mock:server`

which will host the mock data and not result in 404.

File: src/app/routes/features/features.component.ts [hunk 1]
```diff
@@ -17,6 +18,51 @@ export class FeaturesComponent implements OnInit {
   }
 
   ngOnInit() {
+
+      const properties = new Map<string, Property>();
```
Comment (paul58914080) [inline src/app/routes/features/features.component.ts]:
```suggestion
      const properties = new Map<string, Property>();
      properties.set('Sample Property', {
          'name': 'SampleProperty',
          'description': '',
          'type': 'org.ff4j.property.PropertyString',
          'value': 'go!',
          'fixedValues': []
      });
       const initParams = new Map<string, string>();
      initParams.set('expression', 'f1 & f2 | !f1 | f2');
       this.features = [{
          uid: 'Feature_UID',
          description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
          magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
          consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
          pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id
          est laborum.`,
          group: 'Group Name',
          enable: false,
          permissions: [ 'ROLE_ADMIN', 'ROLE_USER', 'ROLE_ADMIN', 'ROLE_USER' ],
          customProperties: properties,
          flippingStrategy: {
              type: 'org.ff4j.strategy.el.ExpressionFlipStrategy',
              initParams: initParams
          }
      },
          {
              uid: 'Feature_UID',
              description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
          magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
          consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
          pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id
          est laborum.`,
              group: 'Group Name',
              enable: false,
              permissions: [ 'ROLE_ADMIN', 'ROLE_USER', 'ROLE_ADMIN', 'ROLE_USER' ],
              customProperties: properties,
              flippingStrategy: {
                  type: 'org.ff4j.strategy.el.ExpressionFlipStrategy',
                  initParams: initParams
              }
          }];
```

is not needed you would have to start with the mock server

File: src/app/shared/components/feature-card/feature-card.component.spec.ts [hunk 2]
```diff
@@ -1,38 +1,53 @@
...
+    group: 'Group Name1',
+    enable: false,
+    permissions: ['ROLE_ADMIN', 'ROLE_USER', 'ROLE_ADMIN', 'ROLE_USER']
+};
 
-  beforeEach(async(() => {
-    TestBed.configureTestingModule({
-      imports: [
-        CommonModule,
+const mockFeature2: Feature = {
```
Comment (paul58914080) [inline src/app/shared/components/feature-card/feature-card.component.spec.ts]: why is mockFeature2 required ?

File: src/app/shared/components/feature-card/feature-card.component.spec.ts [hunk 3]
```diff
@@ -1,38 +1,53 @@
...
+import {Component} from '@angular/core';
+import {Feature} from '../../models/Feature';
+import {initContext, TestContext} from '../../../../testing/test.context';
 import {MatButtonModule, MatCardModule, MatIconModule, MatSlideToggleModule} from '@angular/material';
-import {CUSTOM_ELEMENTS_SCHEMA} from '@angular/core';
 
-describe('FeatureCardComponent', () => {
-  let component: FeatureCardComponent;
-  let fixture: ComponentFixture<FeatureCardComponent>;
+const mockFeature1: Feature = {
```
Comment (paul58914080) [inline src/app/shared/components/feature-card/feature-card.component.spec.ts]: why cant this be within the `TesteeFeatureCardComponent `

File: src/app/shared/components/feature-card/feature-card.component.spec.ts (hunk 2 above)
Comment (J-DK) [reply src/app/shared/components/feature-card/feature-card.component.spec.ts]: This is to test the **on change should apply** at Line #48. Please let me know if you feel it is redundant.

File: src/app/routes/features/features.component.ts (hunk 1 above)
Comment (J-DK) [reply src/app/routes/features/features.component.ts]: In fact, I tried but I still got the error. I ll check once more and remove this.

File: src/app/shared/components/feature-card/feature-card.component.spec.ts (hunk 3 above)
Comment (J-DK) [reply src/app/shared/components/feature-card/feature-card.component.spec.ts]: As I have declared two mocks, I kept it outside the component. If the second mock feature is not needed, I ll place it inside the component

**Comment by paul58914080** [comment]
@J-DK thank you for your contribution

**Pull request merged**